sklearn/linear_model/base.py

```diff
@@ -363,16 +376,29 @@ def fit(self, X, y):
         y : numpy array of shape [n_samples, n_targets]
             Target values
 
+        sample_weight : float or numpy array of shape [n_samples]
```

Comment:
From some quick experiments, weighting all samples by an equal amount does not change anything, as I expected. So I don't think we need to support a scalar sample_weight.